### PR TITLE
executor: support batch-get cluster index & add test case

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Problem Summary:

support batch get on cluster index

and add test-case to cover unique-key point/batch get

### What is changed and how it works?

Proposal: https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-cluster-index.md

What's Changed, How it Works:

support batch get on cluster index

and add test-case to cover unique-key point/batch get

### Related changes

- n/a

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- n/a

### Release note 

- No release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/17858)

### CODE REVIEW

**Comment by lysu** [comment]
need merge https://github.com/pingcap/tidb/pull/17721 firstly~

File: executor/builder.go [hunk 1]
```diff
@@ -3044,19 +3045,39 @@ func (b *executorBuilder) buildBatchPointGet(plan *plannercore.BatchPointGetPlan
 		b.hasLock = true
 	}
 	var capacity int
-	if plan.IndexInfo != nil {
+	if plan.IndexInfo != nil && (!plan.TblInfo.IsCommonHandle || !plan.IndexInfo.Primary) {
```
Comment (coocood) [inline executor/builder.go:3048]: The condition is too long, need a named variable to make it easier to read.

**Comment by coocood** [comment]
LGTM

**Pull request merged**